enhancements/cluster-api/cluster-autoscaler-integration-with-openshift-cluster-api.md

```diff
@@ -0,0 +1,709 @@
+determining the state and health of that scaling group. With this enhancement however, it
+is possible that users might encounter issues when interacting with non-authoritative
+MachineSet resources. To improve the user experience, a conditions field will be added
+to the MachineAutoscaler status field (i.e. `.status.conditions`).
+
+There will be a single condition to indicate the ready status of the MachineAutoscaler. Under
+normal operation this condition will appear similar to this example:
+
+```yaml
+status:
+  conditions:
+    - lastTransitionTime: 2025-01-01T00:00:00Z
+      message: MachineAutoscaler ready for autoscaling
+      reason: MachineAutoscalerReady
+      status: "True"
+      type: Ready
+```
+
+In a case where the user has specified the same MachineSet in multiple MachineAutoscaler resources,
+the condition would appear as follows:
+
+```yaml
+status:
+  conditions:
+    - lastTransitionTime: 2025-01-01T00:00:00Z
+      message: targetScaleRef has multiple MachineAutoscaler owners
+      reason: MachineAutoscalerDuplicateTargetScaleRef
+      status: "False"
+      type: Ready
+```
+
+In this manner, errors in user configuration will be exposed through the MachineAutoscaler
+resource to assist users in diagnosing error states.
+
+#### MachineSet Sync Controller Changes
+
+The behavior of the MachineSet sync controller is another focus of implementation detail.
+Specifically, the MachineSet annotations and the replicas field of the spec
+(i.e. `.spec.replicas`). As mentioned previously, the MachineSet sync controller will
+use the managed fields metadata to know who, or what, has updated a field. Based on the field,
+and the author of the update, the sync controller will either propagate the change
+or synchronize with the authoritative resource.
+
+The updates that the MachineSet sync controller will watch for fall into a few different
+categories:
+
+* `.metadata.annotations` changes will be synced from the authoritative resource to the
+  non-authoritative except in these cases:
+  * The Cluster Autoscaler Operator has added the minimum and maximum size annotations, and ownership
+    annotation to a record. If the sync controller sees an update to these annotations on a
+    non-authoritative resource originating from the Cluster Autoscaler Operator, it will copy
+    that change to the authoritative resource if no MachineAutoscaler is referencing the
+    authoritative resource.
+  * A provider MachineSet controller has added the scale from zero annotations to a
+    non-authoritative record. This occurs when the Cluster API resource is marked as
+    authoritative but the Machine API resource is updated by the provider MachineSet controller.
+    In these cases the scale from zero annotations will be copied to the non-authoritative
+    Cluster API resource. The data from the MachineSet controller is only applied to
+    Machine API resources currently.
```

Comment:
Is there an equivalent of this controller in CAPI? Or, if not, is it on the roadmap? If it is on the roadmap, we will want to ensure these controllers following the same pausing as the rest of the controllers

Reply:
the last time i looked, most if not all of the providers also package a MachineSet actuators, but we actually want to promote a different behavior in the upstream. we want upstream providers to implement infrastructure template controllers to add the capacity information to the status on the infrastructure template, not as annotations on the MachineSet or MachineDeployment.

Reply:
[same as previous reply](https://github.com/openshift/enhancements/pull/1736#discussion_r1954524097), i timelime was never proposed for this as it is "opt-in".

Reply:
correct me if i'm wrong @JoelSpeed , but my understanding is that the MAPI machineset actuators will not be running when CAPI is enabled for a platform. so, there should be no need to reference these annotations?